service/src/main/java/org/broadinstitute/listener/relay/inspectors/HeaderLoggerInspector.java

```diff
@@ -20,27 +24,68 @@ public class HeaderLoggerInspector implements RequestInspector {
   @Override
   public boolean inspectWebSocketUpgradeRequest(
       @NonNull RelayedHttpListenerRequest relayedHttpListenerRequest) {
-
-    logger.debug("WebSocket Upgrade Request: {}", relayedHttpListenerRequest.getUri());
-
-    logHeaders(relayedHttpListenerRequest.getHeaders());
-
+    logRequest(relayedHttpListenerRequest, OffsetDateTime.now(), "WEBSOCKET_UPGRADE_REQUEST");
     return true;
   }
 
   @Override
   public boolean inspectRelayedHttpRequest(
       @NonNull RelayedHttpListenerRequest relayedHttpListenerRequest) {
+    logRequest(relayedHttpListenerRequest, OffsetDateTime.now(), "HTTP_REQUEST");
+    return true;
+  }
+
+  @VisibleForTesting
+  public void logRequest(
+      RelayedHttpListenerRequest relayedHttpListenerRequest,
+      OffsetDateTime requestTimestamp,
+      String prefix) {
+    var headers = relayedHttpListenerRequest.getHeaders();
+    var referer = headers.getOrDefault("Referer", "");
+    var origin = headers.getOrDefault("Origin", "");
+    var ua = headers.getOrDefault("User-Agent", "");
+
+    String endpoint = "";
+    if (relayedHttpListenerRequest.getRemoteEndPoint() != null) {
+      endpoint = relayedHttpListenerRequest.getRemoteEndPoint().toString();
+    }
 
-    logger.debug("HTTP Request: {}", relayedHttpListenerRequest.getUri());
+    var sub = getTokenClaim(relayedHttpListenerRequest.getHeaders(), "sub").orElse("");
+    var idTyp = getTokenClaim(relayedHttpListenerRequest.getHeaders(), "idtyp").orElse("");
+
+    logger.info(
```

Comment:
This is an attempt to get the request logged into something apache-ish.

Reply:
Maybe a comment would be useful that we're trying to mirror the apache format?

Reply:
Hmm not sure I see status code here - we'd need it to detect "Use of service accounts for interactive log on"

Reply:
Yeah, the need for a post response chain complicates things. I'll look into it.

Reply:
I've refactored a bit and moved where we make the logging calls so we can get both the request data + the status codes.